dartagnan/src/main/java/com/dat3m/dartagnan/utils/visualization/ExecutionGraphVisualizer.java

```diff
@@ -34,12 +35,14 @@ public class ExecutionGraphVisualizer {
     private static final Logger logger = LogManager.getLogger(ExecutionGraphVisualizer.class);
 	
     private final Graphviz graphviz;
+    private final Map<Event, String> callStackMapping;
     private BiPredicate<EventData, EventData> rfFilter = (x, y) -> true;
     private BiPredicate<EventData, EventData> coFilter = (x, y) -> true;
     private Map<BigInteger, IExpr> addresses = new HashMap<BigInteger, IExpr>();
 
-    public ExecutionGraphVisualizer() {
+    public ExecutionGraphVisualizer(Map<Event, String> callStackMapping) {
```

Comment:
I think it would be nicer if the map is default-initialized to be an empty `Map` and then you can add a fluent setter `setCallStackMapping` (so it aligns with the way we set the other options).